### PR TITLE
Add auto backport functionality for SQL

### LINKED ISSUE #310: Add auto backport functionality to backport PRs for releases
Coming from meta issue [opensearch-project/opensearch-plugins#108](https://github.com/opensearch-project/opensearch-plugins/issues/108):

- [ ] Add auto-backport PR creation Github action to make sure all the PRs are backported for a particular release. 
- [ ] Add auto-delete for merged auto-backport PRs to clean up the branches created during auto-backport.
- [ ] Add documentation for the same.

Related documentation: https://github.com/opensearch-project/opensearch-plugins/blob/main/BACKPORT.md

### PR DESCRIPTION
**Pull request by davidcui1225**
### Description
Adds auto-backporting functionality for SQL plugin
 
### Issues Resolved
#310 
 
### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass, including unit test, integration test and doctest
- [x] New functionality has been documented.
  - [ ] New functionality has javadoc added
  - [ ] New functionality has user manual doc added
- [x] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Review by joshuali925 — APPROVED** [review]

**Review by ps48 — APPROVED** [review]

**Review by eugenesk24 — APPROVED** [review]
Don't merge without my approval 😄

**Comment by VachaShah** [comment]
Hi @davidcui1225 , this is an old version of the backport, can you please update it and also add auto delete workflow. The details are in issue #310 .

**Pull request merged**
